### PR TITLE
Refactor configuration structure

### PR DESCRIPTION
**Pull request by fmessmer**
related to https://github.com/ipa320/cob_robots/pull/652

### CODE REVIEW

**Comment by fmessmer** [comment]
I consider this done so far....
In https://github.com/ipa320/msh/pull/684, I am testing against all related feature branches...

@ipa-mig @ipa-jba @ipa-fmw 
please review

File: cob_navigation_global/launch/2dnav_linear.launch [hunk 1]
```diff
@@ -3,7 +3,7 @@
 
   <arg name="robot" default="$(optenv ROBOT !!NO_ROBOT_SET!!)"/>
   <arg name="robot_env" default="$(optenv ROBOT_ENV !!NO_ROBOT_ENV_SET!!)"/>
-  <arg name="map" default="$(find cob_default_env_config)/$(arg robot_env)/map.yaml" />
+  <arg name="map" default="$(find cob_default_env_config)/envs/$(arg robot_env)/map.yaml" />
```
Comment (mgruhler) [inline cob_navigation_global/launch/2dnav_linear.launch]: In order to make those launch files more easily usable in the new structure, should we add a launch arg for the env package as well?

File: cob_navigation_global/launch/2dnav_linear.xml [hunk 2]
```diff
@@ -8,7 +8,7 @@
     <remap from="odom" to="base/odometry_controller/odometry"/>
     <!-- remap outputs -->
     <remap from="cmd_vel" to="base/collision_velocity_filter/command"/>
-    <rosparam file="$(find cob_navigation_config)/$(arg robot)/linear/ctrl_params.yaml" command="load" />
+    <rosparam file="$(find cob_navigation_config)/robots/$(arg robot)/nav/linear_ctrl_params.yaml" command="load" />
```
Comment (mgruhler) [inline cob_navigation_global/launch/2dnav_linear.xml]: In order to make those launch files more easily usable in the new structure, should we add a launch arg for the config package as well?

(holds for the other launch files as well, I'm not commenting every occurance now...)

**Review by mgruhler — CHANGES_REQUESTED** [review]
@ipa-fxm what is your opinion?

File: cob_navigation_global/launch/2dnav_linear.launch (hunk 1 above)
Comment (floweisshardt) [reply cob_navigation_global/launch/2dnav_linear.launch]: yes, please use same args as for ipa_navigation

Comment (fmessmer) [reply cob_navigation_global/launch/2dnav_linear.launch]: done

**Review by mgruhler — APPROVED** [review]
merge together with ipa320/cob_environments#120

**Pull request merged**